sdk/src/driftClient.ts

```diff
@@ -467,50 +478,98 @@ export class DriftClient {
 
 				this.userStats = new UserStats({
 					driftClient: this,
-					userStatsAccountPublicKey: getUserStatsAccountPublicKey(
-						this.program.programId,
-						this.authority
-					),
+					userStatsAccountPublicKey: this.getUserStatsAccountPublicKey(),
 					accountSubscription: this.userAccountSubscriptionConfig,
 				});
-			}
-		}
-		this.users.clear();
-		this.createUsers(subAccountIds, this.userAccountSubscriptionConfig);
-		if (this.isSubscribed) {
-			await Promise.all(this.subscribeUsers());
 
-			if (this.userStats) {
 				await this.userStats.subscribe();
 			}
 		}
 
-		this.activeSubAccountId = activeSubAccountId;
-		this.userStatsAccountPublicKey = undefined;
+		this.users.clear();
+
+		await this.addAndSubscribeToUsers(
```

Comment:
why'd you move this out of the `this.isSubscribed` block? feel like it makes sense to only re-subscribe if you were already subscribed

Reply:
i think `isSubscribed` would still be true from a previous wallet connection?

Reply:
this `isSubscribed` doesnt get set in the constructor, just if you call subscribe, so it could not be set? why'd you need to move this re-subscribe outta the if block?